### PR TITLE
[Merged by Bors] - feat: Basic probability lemmas assuming null-measurability

### PR DESCRIPTION
**Pull request by YaelDillies**
and transferring probabiity measures across `ULift`

From LeanCamCombi and PFR

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/Typeclasses.lean [hunk 1]
```diff
@@ -251,35 +251,44 @@ instance isProbabilityMeasureSMul [IsFiniteMeasure μ] [NeZero μ] :
...
+  rw [prob_compl_eq_one_sub₀ hs, tsub_eq_zero_iff_le, one_le_prob_iff]
+
+@[simp] lemma prob_compl_eq_zero_iff (hs : MeasurableSet s) : μ sᶜ = 0 ↔ μ s = 1 :=
+  prob_compl_eq_zero_iff₀ hs.nullMeasurableSet
 #align measure_theory.prob_compl_eq_zero_iff MeasureTheory.prob_compl_eq_zero_iff
 
-@[simp]
-theorem prob_compl_eq_one_iff [IsProbabilityMeasure μ] (hs : MeasurableSet s) :
-    μ sᶜ = 1 ↔ μ s = 0 := by rw [← prob_compl_eq_zero_iff hs.compl, compl_compl]
+@[simp] lemma prob_compl_eq_one_iff (hs : MeasurableSet s) : μ sᶜ = 1 ↔ μ s = 0 := by
```
Comment (ADedecker) [inline Mathlib/MeasureTheory/Measure/Typeclasses.lean:286]: Could you add the `NullMeasurableSet` version of this one?

File: Mathlib/MeasureTheory/Measure/Typeclasses.lean [hunk 2]
```diff
@@ -299,6 +308,17 @@ protected lemma _root_.MeasurableEmbedding.isProbabilityMeasure_comap (hf : Meas
...
+instance isProbabilityMeasure_map_up [IsProbabilityMeasure μ] :
+    IsProbabilityMeasure (μ.map ULift.up) := isProbabilityMeasure_map measurable_up.aemeasurable
+
+instance isProbabilityMeasure_comap_down [IsProbabilityMeasure μ] :
+    IsProbabilityMeasure (μ.comap ULift.down) := by
+  refine isProbabilityMeasure_comap ?_ ?_ $ by
+    rintro s hs
+    change MeasurableSet (ULift.down ⁻¹' (ULift.down '' s))
+    rwa [Set.preimage_image_eq _ ULift.down_injective]
+  all_goals simp [ULift.down_injective]
```
Comment (ADedecker) [inline Mathlib/MeasureTheory/Measure/Typeclasses.lean:320]:
```suggestion
    IsProbabilityMeasure (μ.comap ULift.down) :=
  MeasurableEquiv.ulift.measurableEmbedding.isProbabilityMeasure_comap <| ae_of_all _ <| by
  simp [Function.Surjective.range_eq <| EquivLike.surjective _]
```

**Review by ADedecker — APPROVED** [review]
Thanks!

bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**